subprojects/normalization-java/src/main/java/org/gradle/api/internal/changedetection/state/NormalizedContentInfoCollector.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.changedetection.state;
+
+import org.gradle.internal.hash.FileContentType;
+import org.gradle.internal.hash.FileContentTypeDetectingInputStream;
+import org.gradle.internal.hash.HashCode;
+import org.gradle.internal.hash.StreamHasher;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.util.function.Function;
+
+public class NormalizedContentInfoCollector {
```

Comment:
This could use some Javadoc.

Reply:
It's been removed altogether.  It was an unnecessary complexity.